### PR TITLE
refactor(mme): convert test_mme_app_ue_context to gtest

### LINKED ISSUE #11660: mme: Convert unit tests to use Google Test framework
## Problem
All other C++ unit tests (MME and others) are written using Google Test. The goal is to unify the test framework for all C++ unit tests so that they can be run in the same way.

## Exit Criteria
Migrate the test to be run with gtest OR figure out a strategy on running this test easily with `bazel test //...` command

## Tests that need refactoring
- [x] lte/gateway/c/core/oai/test/mobility_client/test_mobility_client.cpp
- [x] lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.c

### PR DESCRIPTION
**Pull request by LKreutzer**
Signed-off-by: Lars Kreutzer <lars.kreutzer@tngtech.com>

## Summary

- Convert test_mme_app_ue_context to gtest (see https://github.com/magma/magma/issues/11660).
- Converted test to valid C++ code because gtest is designed for C++ only.
- Note: For the review please change the file extension locally to `.c` in order for the diff view to display properly.

## Test Plan

- Run `make test_oai`
- Run `bazel test //lte/gateway/c/core/oai/test/mme_app_task:mme_app_ue_context_test` (on the `electronjoe/pr-giant-bazel-mme-rebase`)
- CI

## Additional Information

- [ ] This change is backwards-breaking

@electronjoe FYI static linking is now not needed with bazel, the following target works:
```
cc_test(
   name = "mme_app_ue_context_test",
   srcs = ["test_mme_app_ue_context.cpp"],
   deps = [
       "//lte/gateway/c/core:core",
       "@com_google_googletest//:gtest_main",
   ],
)
```

### CODE REVIEW

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.cpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+#define TEST_CASE_COMMON_CONVERT_MAX 10
+
+/**
+ * @brief mme_app_convert_imsi_to_imsi_mme: converts the imsi_t struct to the
+ * imsi mme struct
+ * @param imsi_dst
+ * @param imsi_src
+ */
+// TODO: (amar) This and below functions are only used in testing possibly move
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.cpp]: I think this comment is outdated, can you remove it as well?

**Review by ardzoht — APPROVED** [review]
lgtm with minor comment

**Comment by electronjoe** [comment]
> FYI static linking is now not needed with bazel, the following target works:

Thanks! Yeah I had thought I removed all the linkstatic from cc_tests in my branch? Did I leave some dangling? I'll go grep.

**Review by electronjoe — COMMENTED** [review]
How annoying is it to address the cpplint findings in the file?

E.g. 

> Using C-style cast. Use const_cast<char*>(...) instead [readability/casting] [4]

and kin?

I'll add that if that makes this PR more difficult to diff, this could be a follow up PR.

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.cpp (hunk 1 above)
Comment (LKreutzer) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.cpp]: Done!

**Comment by LKreutzer** [comment]
> > FYI static linking is now not needed with bazel, the following target works:
> 
> Thanks! Yeah I had thought I removed all the linkstatic from cc_tests in my branch? Did I leave some dangling? I'll go grep.

Seems to have been the only `cc_test` with `linkstatic`.

> How annoying is it to address the cpplint findings in the file?
> 
> E.g.
> 
> > Using C-style cast. Use const_cast<char*>(...) instead [readability/casting] [4]
> 
> and kin?
>
> I'll add that if that makes this PR more difficult to diff, this could be a follow up PR.

Done and pushed!

**Comment by themarwhal** [comment]
The mme unit test is fixed on master, so rebasing should do the trick :)

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.cpp [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  ASSERT_EQ(mme_app_imsi_compare(&imsi_mme_a, &imsi_mme_a), true);
+  ASSERT_EQ(mme_app_imsi_compare(&imsi_mme_b, &imsi_mme_a), false);
+  ASSERT_EQ(mme_app_imsi_compare(&imsi_mme_a, &imsi_mme_a), true);
+}
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  OAILOG_INIT("MME", OAILOG_LEVEL_DEBUG, MAX_LOG_PROTOS);
+  return RUN_ALL_TESTS();
+}
```
Comment (themarwhal) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.cpp:280]: do you know if this is needed? (what does OAILOG_INIT do?)

Comment (ardzoht) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_ue_context.cpp:280]: @themarwhal this is needed as there are some -ugly- tangled dependencies within the codebase that expect the log module to be initialized.

**Pull request merged**